public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx

```diff
@@ -0,0 +1,340 @@
+import { RESTORE_OPTIONS } from "../../../../models/interfaces";
+import { getErrorMessage } from "../../../../utils/helpers";
+import { IndexItem } from "../../../../../models/interfaces";
+import { CatRepository, GetSnapshot } from "../../../../../server/models/interfaces";
+import CustomLabel from "../../../../components/CustomLabel";
+import SnapshotRestoreAdvancedOptions from "../SnapshotRestoreAdvancedOptions";
+import SnapshotRestoreOption from "../SnapshotRestoreOption";
+import SnapshotRenameOptions from "../SnapshotRenameOptions";
+import AddPrefixInput from "../AddPrefixInput";
+import RenameInput from "../RenameInput";
+import SnapshotIndicesInput from "../SnapshotIndicesInput";
+import { ERROR_PROMPT } from "../../../CreateSnapshotPolicy/constants";
+
+interface RestoreSnapshotProps {
+  snapshotManagementService: SnapshotManagementService;
+  indexService: IndexService;
+  onCloseFlyout: () => void;
+  restoreSnapshot: (snapshotId: string, repository: string, options: object) => void;
+  snapshotId: string;
+}
+
+interface RestoreSnapshotState {
+  indexOptions: EuiComboBoxOptionOption<IndexItem>[];
+  selectedIndexOptions: EuiComboBoxOptionOption<IndexItem>[];
+  renameIndices: string;
+  prefix: string;
+  renamePattern: string;
+  renameReplacement: string;
+  listIndices: boolean;
+
+  repositories: CatRepository[];
+  selectedRepoValue: string;
+
+  snapshot: GetSnapshot | null;
+  snapshotId: string;
+  restoreSpecific: boolean;
+  partial: boolean;
+
+  repoError: string;
+  snapshotIdError: string;
+}
+
+export default class RestoreSnapshotFlyout extends Component<RestoreSnapshotProps, RestoreSnapshotState> {
+  static contextType = CoreServicesContext;
+  constructor(props: RestoreSnapshotProps) {
+    super(props);
+    this.state = {
+      indexOptions: [],
+      selectedIndexOptions: [],
+      renameIndices: "add_prefix",
+      prefix: "",
+      renamePattern: "",
+      renameReplacement: "",
+      listIndices: false,
+      repositories: [],
+      selectedRepoValue: "",
+      snapshot: null,
+      snapshotId: "",
+      restoreSpecific: false,
+      partial: false,
+      repoError: "",
+      snapshotIdError: "",
+    };
+  }
+
+  async componentDidMount() {
+    await this.getRepos();
+    await this.getIndexOptions();
+  }
+
+  onClickAction = () => {
+    const { restoreSnapshot, snapshotId } = this.props;
+    const {
+      selectedRepoValue,
+      restoreSpecific,
+      selectedIndexOptions,
+      indexOptions,
+      snapshot,
+      renameIndices,
+      prefix,
+      renamePattern,
+      renameReplacement,
+    } = this.state;
+    const { add_prefix } = RESTORE_OPTIONS;
+    const selectedIndices = selectedIndexOptions.map((option) => option.label).join(",");
+    const allIndices = indexOptions.map((option) => option.label).join(",");
+    const pattern = renameIndices === add_prefix ? "(?<![^ ])(?=[^ ])" : renamePattern;
```

Comment:
nit: Add explanation on why this regex works 😃

Reply:
Left a TODO to get back to.